### PR TITLE
Samples:Host Exerciser Enhancement

### PR DESCRIPTION
**Pull request by ramyavex**
Modified the source to access the sysfs path for formats and events
using glob instead of udev library.
With udev library in Centos8 machine not able to get events and format attributes for dfl fme.  

Signed-off-by: ramyavv <ramyax.venkatesha@intel.com>

### CODE REVIEW

**Review by r-rojo — COMMENTED** [review]
Seems like parse_perf_format and parse_perf_event are very similar.
We should avoid code duplication as much as possible.
What's different about them? Is it mainly the  attribute directory (glob pattern) and the regex pattern?
If that's the case, perhaps you can refactor to one function and parameterize both the glob pattern and the regex pattern.

File: samples/fpgaperf_counter/fpgaperf_counter.c [hunk 1]
```diff
@@ -104,142 +104,142 @@ STATIC fpga_result fpga_perf_check_and_lock(fpga_perf_counter *fpga_perf)
...
 		if (!fpga_perf->format_type) {
-			fpga_perf->num_format = 0;
-			OPAE_ERR("Failed to allocate Memory");
-			return FPGA_NO_MEMORY;
+			fpga_perf->format_type = calloc(fpga_perf->num_format,
+						sizeof(perf_format_type));
+			if (!fpga_perf->format_type) {
+				fpga_perf->num_format = 0;
+				OPAE_ERR("Failed to allocate Memory");
+				return FPGA_NO_MEMORY;
```
Comment (tswhison) [inline samples/fpgaperf_counter/fpgaperf_counter.c:151]: Need a
```c
globfree(&pglob);
```
here as well?

File: samples/fpgaperf_counter/fpgaperf_counter.c [hunk 2]
```diff
@@ -104,142 +104,142 @@ STATIC fpga_result fpga_perf_check_and_lock(fpga_perf_counter *fpga_perf)
 	return FPGA_OK;
 }
 
-/* parse the each format and get the shift val */
-STATIC fpga_result parse_perf_format(struct udev_device *dev, fpga_perf_counter *fpga_perf)
+/* parse the each format and get the shift val
+ * parse the events for the particular device directory */
+STATIC fpga_result parse_perf_attributes(struct udev_device *dev,
+			fpga_perf_counter *fpga_perf, const char *attr)
```
Comment (r-rojo) [inline samples/fpgaperf_counter/fpgaperf_counter.c:110]: Shouldn't the regex pattern be parameterized here too?

File: samples/fpgaperf_counter/fpgaperf_counter.c [hunk 3]
```diff
@@ -275,11 +275,11 @@ STATIC fpga_result fpga_perf_events(char* perf_sysfs_path, fpga_perf_counter *fp
 		PARSE_MATCH_INT(ptr, 0, fpga_perf->type, 10);
 
 	/* parse the format value */
-	ret = parse_perf_format(dev, fpga_perf);
+	ret = parse_perf_attributes(dev, fpga_perf, "format");
 	if (ret != FPGA_OK)
 		goto out;
 	/* parse the event value */
-	ret = parse_perf_event(dev, fpga_perf);
+	ret = parse_perf_attributes(dev, fpga_perf, "events");
```
Comment (r-rojo) [inline samples/fpgaperf_counter/fpgaperf_counter.c:282]: Should the regex pattern used the same for calling this on line 278 and 282? Shouldn't the pattern depend on the attribute kind or is only one regex pattern valid?

**Review by r-rojo — APPROVED** [review]
We can refactor later.

**Review by r-rojo — APPROVED** [review]

**Pull request merged**